### DISCUSSION TITLE
Ping collision?

### DISCUSSION THREAD

**Discussion by drHuangMHT** [libp2p/rust-libp2p#3329] [General]
In ping protocol one peer sends a ping and expects a pong from the other peer, but what if both peers send a ping at the same time? Will the expected pong be ping from the other peer and trigger a mismatch?

**Comment by mxinden** [comment]
Pings are send in separate streams in isolation of each other, thus there is no mismatch.

**Reply by drHuangMHT** [reply]
Thanks for the instant reply! I'm just getting started on libp2p, not very clear about what happens behind the scenes. So may you recommend some sources to learn from?

**Reply by mxinden** [reply]
https://libp2p.io/ links to everything, e.g. docs, talks, ...
